sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_pyamqp_transport.py

```diff
@@ -0,0 +1,561 @@
+        :keyword int idle_timeout: Required.
+        :keyword Dict properties: Required.
+        :keyword int remote_idle_timeout_empty_frame_send_ratio: Required.
+        :keyword error_policy: Required.
+        :keyword bool debug: Required.
+        :keyword str encoding: Required.
+        """
+        endpoint = kwargs.pop("endpoint")
+        host = kwargs.pop("host")  # pylint:disable=unused-variable
+        auth = kwargs.pop("auth")  # pylint:disable=unused-variable
+        network_trace = kwargs.pop("debug")
+        return Connection(endpoint, network_trace=network_trace, **kwargs)
+
+    @staticmethod
+    def close_connection(connection):
+        """
+        Closes existing connection.
+        :param connection: uamqp or pyamqp Connection.
+        """
+        connection.close()
+
+    @staticmethod
+    def get_connection_state(connection):
+        """
+        Gets connection state.
+        :param connection: uamqp or pyamqp Connection.
+        """
+        return connection.state
+
+    @staticmethod
+    def create_send_client(*, config, **kwargs):  # pylint:disable=unused-argument
+        """
+        Creates and returns the uamqp SendClient.
+        :param ~azure.eventhub._configuration.Configuration config: The configuration.
+
+        :keyword str target: Required. The target.
+        :keyword JWTTokenAuth auth: Required.
+        :keyword int idle_timeout: Required.
+        :keyword network_trace: Required.
+        :keyword retry_policy: Required.
+        :keyword keep_alive_interval: Required.
+        :keyword str client_name: Required.
+        :keyword dict link_properties: Required.
+        :keyword properties: Required.
+        """
+
+        target = kwargs.pop("target")
+        # TODO: extra passed in to pyamqp, but not used. should be used?
+        msg_timeout = kwargs.pop("msg_timeout")  # pylint: disable=unused-variable  # TODO: not used by pyamqp?
+
+        return SendClient(
+            config.hostname,
+            target,
+            custom_endpoint_address=config.custom_endpoint_address,
+            connection_verify=config.connection_verify,
+            transport_type=config.transport_type,
+            http_proxy=config.http_proxy,
+            **kwargs,
+        )
+
+    @staticmethod
+    def send_messages(producer, timeout_time, last_exception, logger):
+        """
+        Handles sending of event data messages.
+        :param ~azure.eventhub._producer.EventHubProducer producer: The producer with handler to send messages.
+        :param int timeout_time: Timeout time.
+        :param last_exception: Exception to raise if message timed out. Only used by uamqp transport.
+        :param logger: Logger.
+        """
+        # pylint: disable=protected-access
+        producer._open()
+        timeout = timeout_time - time.time() if timeout_time else 0
+        producer._handler.send_message(producer._unsent_events[0], timeout=timeout)
+        producer._unsent_events = None
```

Comment:
This seems to be resetting `unsent_events` even if there might have been more than one event there?

Reply:
ah will fix this, to set to None only if the list is >1

Reply:
Gotcha - maybe we should add a comment to this code describing that.
It probably wouldn't hurt to refactor at some point - but I don't think this is the time :)